src/option_parser.c

```diff
@@ -170,36 +170,77 @@ int string_parse_sepcolor(const void *data, const char *s, void *ret)
 {
         LOG_D("parsing sep_color");
         struct separator_color_data *sep_color = (struct separator_color_data*) ret;
+        struct color invalid = COLOR_UNINIT;
 
         enum separator_color type;
         bool is_enum = string_parse_enum(data, s, &type);
         if (is_enum) {
                 sep_color->type = type;
-                g_free(sep_color->sep_color);
-                sep_color->sep_color = NULL;
+                sep_color->color = invalid;
                 return true;
         } else {
                 if (STR_EMPTY(s)) {
                         LOG_W("Sep color is empty, make sure to quote the value if it's a color.");
                         return false;
                 }
-                if (s[0] != '#') {
-                        LOG_W("Sep color should start with a '#'");
-                        return false;
-                }
-                if (strlen(s) < 4) {
-                        LOG_W("Make sure the sep color is formatted correctly");
-                        return false;
-                }
-                // TODO add more checks for if the color is valid
 
-                sep_color->type = SEP_CUSTOM;
-                g_free(sep_color->sep_color);
-                sep_color->sep_color = g_strdup(s);
-                return true;
+                if (string_parse_color(NULL, s, &sep_color->color)) {
+                        sep_color->type = SEP_CUSTOM;
+                        return true;
+                }
         }
+        return false;
 }
 
+#define UINT_MAX_N(bits) ((1 << bits) - 1)
+
+// Parse a #RRGGBB[AA] string
+int string_parse_color(const void *data, const char *s, void *ret)
+{
+        if (STR_EMPTY(s) || *s != '#') {
```

Comment:
I know it was basically this way already before your changes, but `dunst.5` states, that named colors like "Yellow" are supported, which doesn't seem to be the case anymore (I don't know since when). Either it is removed from the documentation (breaking change, should be added - retroactively - to `CHANGELOG.md`) or it is possible to somehow get it back to work (starting point could be [this list](https://gitlab.freedesktop.org/xorg/app/rgb/raw/master/rgb.txt)).

Reply:
Mmm, didn't know about that. I moved the parsing code from draw.c. maybe we can add color names in another pr.

Did you think about embedding an array with all the info from that file?